### PR TITLE
chore(build): Use https://github.com/golangci/golangci-lint-action

### LINKED ISSUE #651: Use https://github.com/golangci/golangci-lint-action to speed up lint
The lint dependencies are not cached properly but there is an official GitHub action from the golangci-lint folks that is optimized for performance in GitHub.

Conceivably we could also enable this to work on Windows gracefully.

### PR DESCRIPTION
**Pull request by gdamore**
Fixes #651

### CODE REVIEW

**Comment by gdamore** [comment]
This isn't right yet.

**Comment by gdamore** [comment]
Note:  Per recommendations, we run the lint actions in their own job, to improve parallelism.

**Comment by gdamore** [comment]
Git hub actions was confused by this commit, as I was changing workflow and it was busted for a bit.  Those 6 expected actions will never complete.

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -9,16 +9,14 @@ on:
     branches:
       - master
 
-
 jobs:
-
-  test:
+   test:
```
Comment (xcoulon) [inline .github/workflows/ci-build.yml:13]: the number of spaces (3) is weird. Would you mind fixing it?

Comment (gdamore) [reply .github/workflows/ci-build.yml:13]: ok.  I didn't even realize this changed.

**Comment by gdamore** [comment]
Should be ready to go now.

**Comment by xcoulon** [comment]
> Should be ready to go now.

yes, I'm waiting for the jobs to complete and I'll merge it ;)

**Comment by xcoulon** [comment]
there's something weird with the checks, though: there are 6 of them stuck with the following status `expected -- waiting for status to be reported`. Is it what you meant in https://github.com/bytesparadise/libasciidoc/pull/652#issuecomment-650424681 as well?

**Comment by gdamore** [comment]
The six expected checks are confusion in GitHub actions... I effectively renamed some actions partway through this PR (added "Test" prefix), and that has caused GitHub some grief.

Part of this was probably also exacerbated by a syntax error in the YAML that caused the original tests to never finish, when I was working on it to begin with.  (Unfortunately in order to start this process and test it, I had to create a PR.  It wasn't right to begin with.)

If you want we can decline this PR, and I can start a new one which should be clean from the beginning (none of the screwy history breaking actions.)

**Comment by xcoulon** [comment]
> The six expected checks are confusion in GitHub actions... I effectively renamed some actions partway through this PR (added "Test" prefix), and that has caused GitHub some grief.
> 
> Part of this was probably also exacerbated by a syntax error in the YAML that caused the original tests to never finish, when I was working on it to begin with. (Unfortunately in order to start this process and test it, I had to create a PR. It wasn't right to begin with.)
> 
> If you want we can decline this PR, and I can start a new one which should be clean from the beginning (none of the screwy history breaking actions.)

instead of declining it immediately, could you open a new PR with the commit you already have here to see if it works?

**Comment by gdamore** [comment]
So I'm pretty sure that you have branch permissions set up to require certain status checks pass, and that this change breaks those settings.  I think you should review a bit more closely, because I'm sure you want to add the Lint rules, and you should add the Test prefix to the other rules.

**Comment by xcoulon** [comment]
> So I'm pretty sure that you have branch permissions set up to require certain status checks pass, and that this change breaks those settings. I think you should review a bit more closely, because I'm sure you want to add the Lint rules, and you should add the Test prefix to the other rules.

oooh, indeed! My apologies, I totally forgot about those branch requirements! It's all good now, so I will merge this PR and close #654 if it's ok for you

**Pull request merged**